raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -195,10 +200,11 @@ public void initialize(OffsetAndEpoch logEndOffsetAndEpoch) throws IllegalStateE
                 logContext
             );
         } else if (election.hasVoted()) {
-            initialState = new VotedState(
+            initialState = new UnattachedState(
```

Comment:
This validates that merging these two states is correct. Not much state difference and you didn't have to change `pollUnattached`.